### PR TITLE
Action Items story. Bug fixes and Enhancement done.

### PR DESCRIPTION
**Pull request by pant-manas**
Worked on bug fixes and enhancements

In bugs there were bugs of onsubmit message and carlisting page coming from middle. Fixed those

In enhancements the error messages were to be changed in form according to the input typed in. 
Like if email input is empty, Email required
If email is incorrect
Invalid Email
Similarly for contact field, name field and address field

https://abhirams.atlassian.net/browse/P3ECWFC-77

https://abhirams.atlassian.net/browse/P3ECWFC-76

Also worked on enhancements
https://abhirams.atlassian.net/browse/P3ECWFC-82
https://abhirams.atlassian.net/browse/P3ECWFC-83

### CODE REVIEW

**Comment by pashedin** [comment]
`spill over work from sprint 1`
`Worked on the remaining part of Sprint 1`
these two statements donot give me any idea o what I to expect in the code, mention or list the features as well.

**Comment by pant-manas** [comment]
> `spill over work from sprint 1` `Worked on the remaining part of Sprint 1` these two statements donot give me any idea o what I to expect in the code, mention or list the features as well.

@pashedin  there were just enhancements and bugs to fix. So thats why just added the links for the same things. 

In bugs there were bugs of onsubmit message and carlisting page coming from middle
In enhancements the error messages were to be changed in form according to the input typed in. Like if email input is empty, Email required
If email is incorrect 
Invalid Email

File: src/pages/TestDrive/TestDrive.tsx [hunk 1]
```diff
@@ -1,32 +1,27 @@
-import { Button, Container, Form, Card, Modal } from "react-bootstrap";
-import { useLocation, useParams, Link, useNavigate } from "react-router-dom";
-import React, { useEffect, useState } from "react";
+import { Button, Container, Form, Card, Modal, FormGroup } from "react-bootstrap";
+import React from "react";
 import "./TestDrive.scss";
 import { contactpattern, emailpattern } from "../../constants";
+import { useForm } from "react-hook-form";
```
Comment (pashedin) [inline src/pages/TestDrive/TestDrive.tsx:5]: here you are utilizing react-hook form
This PR is removing it Please co-ordiante with Kartik and Pratik
https://github.com/DteTRPod3/TRTraining_FinalAssignment/pull/26/files

File: src/pages/TestDrive/TestDrive.tsx [hunk 2]
```diff
@@ -35,85 +30,76 @@ function TestDrive() {
...
+                    minLength: {
+                      value: 10,
+                      message: "Mobile number cannot be less than 10 digits",
+                    },
+                    maxLength: {
+                      value: 10,
+                      message: "Mobile number cannot be more than 10 digits",
+                    },
+                  })}
                 />
```
Comment (pashedin) [inline src/pages/TestDrive/TestDrive.tsx]: why change it back to number, type tel was good, infact much better.
`<input type="tel" placeholder="Mobile number" {...register("Mobile number", {required: true, minLength: 6, maxLength: 12})} />`
https://react-hook-form.com/form-builder/

File: src/pages/TestDrive/__test__/TestDrive.test.tsx [hunk 3]
```diff
@@ -52,7 +52,7 @@ describe("Test Drive Form", () => {
     });
 
     expect(screen.getByPlaceholderText(/enter contact\.\.\./i)).toHaveValue(
-      "1111111111"
+      1111111111
```
Comment (pashedin) [inline src/pages/TestDrive/__test__/TestDrive.test.tsx]: For test cases add believable input 😄 
1234567890 is still fine, or I would say add a random but believable input 9923876104.
also add the test cases where we test that on submitting wrong value an error is thrown.

**Review by pashedin — CHANGES_REQUESTED** [review]

**Comment by pashedin** [comment]
> > `spill over work from sprint 1` `Worked on the remaining part of Sprint 1` these two statements donot give me any idea o what I to expect in the code, mention or list the features as well.
> 
> @pashedin there were just enhancements and bugs to fix. So thats why just added the links for the same things.
> 
> In bugs there were bugs of onsubmit message and carlisting page coming from middle In enhancements the error messages were to be changed in form according to the input typed in. Like if email input is empty, Email required If email is incorrect Invalid Email

update the description and the title with this info 😄

**Comment by pant-manas** [comment]
> Referen

@pashedin  Commited the changes you suggested.
Added test case for error messages
Added tel for attribute for contact
Completed the bugs and enhancements

**Review by pashedin — APPROVED** [review]
Nice quick work man 👍

**Pull request merged**